app/lib/audit/models.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+import '../account/models.dart';
+import '../shared/datastore.dart' as db;
+import '../shared/utils.dart' show createUuid;
+
+@db.Kind(name: 'AuditLogRecord', idType: db.IdType.String)
+class AuditLogRecord extends db.ExpandoModel<String> {
+  @db.DateTimeProperty(required: true)
+  DateTime created;
+
+  @db.DateTimeProperty()
+  DateTime expires;
+
+  @db.StringProperty(required: true)
+  String kind;
+
+  @db.StringProperty(required: true)
+  String agent;
```

Comment:
```suggestion
  /// [User.id] of the user who initiated / authorized the recorded action.
  ///
  /// This is a UUIDv4. If the user has been deleted, it is possible that this
  /// property may not match any [User] entity.
  @db.StringProperty(required: true)
  String agent;
```

TODO: Double check that `[User.id]` is the correct identifier.